### PR TITLE
Strip whitespace from commit SHA

### PR DESCRIPTION
**Pull request by sethmlarson**
Closes https://github.com/python/release-tools/issues/90

### CODE REVIEW

File: sbom.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def get_release_tools_commit_sha() -> str:
     stdout = subprocess.check_output(
         ["git", "rev-parse", "--prefix", git_prefix, "HEAD"],
         cwd=git_prefix
-    ).decode("ascii")
+    ).decode("ascii").strip()
     assert re.match(r"^[a-f0-9]{40,}$", stdout)
```
Comment (ezio-melotti) [inline sbom.py:108]: This `assert` is already ensuring that the `stdout` only contains the hex SHA, with no spaces.  In #90 you mentioned that sometimes there might be a newline attached, but from the report it's not clear if you saw this `assert` fail or if there was some other symptom.
Adding the `.strip()` makes no harm, so this PR is ok, but if you didn't see the assert fail, then the newline might be coming from some other place.

**Review by ezio-melotti — APPROVED** [review]

File: sbom.py (hunk 1 above)
Comment (sethmlarson) [reply sbom.py:108]: I wondered about the too, @Yhg1s  did you run into this during the 3.12.2 release?

Comment (Yhg1s) [reply sbom.py:108]: Nope, I ran the release with unmodified release-tools sources, and this assert didn't trigger.

Comment (sethmlarson) [reply sbom.py:108]: Dang, I feel like I've seen a ghost. Had no idea that `$` had this behavior, now I wonder how many subtle bugs are out there.

> `$`: Matches the end of the string or just before the newline at the end of the string

**Pull request merged**